Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1

```diff
@@ -92,6 +93,25 @@ function Write-DTLServiceDeprecationMessageIfRequired
     }
 }
 
+function Publish-Azure-Telemetry
+ {
+   param([object] $deploymentResponse, [string] $jobId)
+    if($deploymentResponse){
+       $jsonString = -join("{" , 
+       "`"IsAzureVm`" : `"$($deploymentResponse.IsAzureVm)`"" , 
+       "," , 
+       "`"VmUuidHash`" : `"$($deploymentResponse.VmUuidHash)`"" , 
+       "," , 
+       "`"TelemetryError`" : `"$($deploymentResponse.TelemetryError)`"" ,
```

Comment:
TelemetryError can be empty if the execution was successful and viceversa if there was error. Check for the members before you reference. If PS is run in StrictMode referencing can fail.

Reply:
Done. Now Checking field before using it.